lstchain/reco/r0_to_dl1.py

```diff
@@ -267,6 +310,167 @@ def get_dl1(
     return dl1_container
 
 
+def get_dl1_lh_fit(
+    calibrated_event,
+    subarray,
+    geometry,
+    telescope_id,
+    dl1_container,
+    normalized_pulse_template,
+    pedestal_std,
+    custom_config
+):
+    """
+    Return a DL1ParametersContainer of extracted features from a calibrated
+    event. The features are extracted by maximizing an image likelihood
+    function over pixels and time samples. The model considers an asymmetric
+    2D Gaussian distribution of the charge and a linear temporal model. The
+    spatio-temporal image model is then compared to the signal vs time in each
+    pixel while taking into account the response of the instrument from
+    calibration.
+    The DL1ParametersContainer needs to contain a first features estimation
+    as seed for the likelihood fit.
+    Parameters
+    ----------
+    calibrated_event: ctapipe event container
+    subarray: `ctapipe.instrument.subarray.SubarrayDescription`
+    geometry: `ctapipe.instrument.CameraGeometry`
+    telescope_id: `int`
+    dl1_container: DL1ParametersContainer
+    normalized_pulse_template: NormalizedPulseTemplate
+    pedestal_std: array-like or None
+        Pedestal standard deviation from interleaved events (WIP) from DL1 file
+        or None to use event-wise signal-less pixels standard deviation
+    custom_config: path to a configuration file
+        configuration used for the likelihood fit
+    Returns
+    -------
+    DL1ParametersContainer
+    """
+    lh_fit_config = custom_config['lh_fit_config']
+
+    waveform = calibrated_event.r1.tel[telescope_id].waveform
+
+    time_shift = None
+    if dl1_container.mc_type == -9999:  # Find time shift correction for data
```

Comment:
This is the default value, which is not updated for real observation data file. In case of MC is is dependent of the particle type. Previously I passed the is_simu flag as an additional parameter.
Switching to :
if calibrated_event.simulation is None:
as it may be easier to understand.